src/python/grpcio_tests/tests/observability/_observability_test.py

```diff
@@ -157,13 +162,29 @@ def testRecordUnaryUnary(self):
             exporter=self.test_exporter
         ):
             self._start_server()
-            self.unary_unary_call()
+            unary_unary_call(port=self._port)
 
         self.assertGreater(len(self.all_metric), 0)
         self.assertGreater(len(self.all_span), 0)
         self._validate_metrics(self.all_metric)
         self._validate_spans(self.all_span)
 
+    def testContextPropagation(self):
+        # Sends two RPCs, one from gRPC client and the other from gRPC server.
+        # Verify that the trace_id was propagated to the 2nd RPC.
+        self._set_config_file(_VALID_CONFIG_TRACING_ONLY)
+        with grpc_observability.GCPOpenCensusObservability(
+            exporter=self.test_exporter
+        ):
+            self._start_server(port=PORT)
+            unary_unary_call(port=PORT, metadata=TRIGGER_RPC_METADATA_TUPLE)
+
+        # 2 of each for ["Recv", "Sent", "Attempt"]
+        self.assertEqual(len(self.all_span), 6)
```

Comment:
I think it would be useful to have a test that records one more RPC in the client -> server -> server chain. The tricky part is making sure that the context from an incoming request propagates to an outgoing request. While this _mostly_ does that, it's possible for an implementation to record an outgoing RPC as belonging to a trace (since the first server is responsible for sending this info out) but for the RPC to not actually carry the trace information, resulting in the second server not receiving this trace.

The most straightforward way to to do this would be to have the server be sort of "recursive" by sending a request to itself, but this runs the risk of causing false positive tests; if the server thread that receives the response is the same as the server thread that sends the request, then the test can pass trivially without any trace context propagation functionality happening at all. I would suggest either going multiprocess or having multiple servers in the same process, each with their own thread pool.